auth/src/main/java/org/jboss/pnc/bacon/auth/DirectKeycloakClientImpl.java

```diff
@@ -0,0 +1,141 @@
+package org.jboss.pnc.bacon.auth;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.ObjectMapper;
+import com.mashape.unirest.http.Unirest;
+import lombok.extern.slf4j.Slf4j;
+import org.jboss.pnc.bacon.auth.model.Credential;
+import org.jboss.pnc.bacon.auth.model.KeycloakResponse;
+import org.jboss.pnc.bacon.auth.spi.KeycloakClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * Authenticates to Keycloak using direct flow!
+ */
+@Slf4j
+public class DirectKeycloakClientImpl implements KeycloakClient {
+
+    private static void setupUnirest() {
+        // Only one time
+        Unirest.setObjectMapper(new ObjectMapper() {
+            private com.fasterxml.jackson.databind.ObjectMapper jacksonObjectMapper
+                    = new com.fasterxml.jackson.databind.ObjectMapper();
+
+            public <T> T readValue(String value, Class<T> valueType) {
+                try {
+                    return jacksonObjectMapper.readValue(value, valueType);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            public String writeValue(Object value) {
+                try {
+                    return jacksonObjectMapper.writeValueAsString(value);
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+    }
+
+    public DirectKeycloakClientImpl() {
```

Comment:
Could `setupUnirest()` be done in static block? Just in case someone creates many instances of this class